clusterman/autoscaler/pool_manager.py

```diff
@@ -203,6 +203,35 @@ def get_node_metadatas(self, state_filter: Optional[Collection[str]] = None) ->
             for instance_metadata in group.get_instance_metadatas(state_filter)
         ]
 
+    def terminate_expired_orphan_instances(self, dry_run: bool = False) -> None:
+        if dry_run:
+            logger.warning('Running in "dry-run" mode; cluster state will not be modified')
+
+        group_id_to_instance_ids = self.get_expired_orphan_instances()
+        logger.info(f"Terminating expired orphan instances: \n{group_id_to_instance_ids}")
+
+        if dry_run:
+            return
+
+        for group_id in group_id_to_instance_ids:
+            self.resource_groups[group_id].terminate_instances_by_id(group_id_to_instance_ids[group_id])
+
+    def get_expired_orphan_instances(self, timeout: int = 1800) -> Mapping[str, List[str]]:
+        groups = self.resource_groups.values()
+        node_metadatas = self.get_node_metadatas(AWS_RUNNING_STATES)
+
+        group_id_to_instance_ids: Mapping[str, List[str]] = defaultdict(list)
+
+        for group in groups:
+            for metadata in node_metadatas:
+                if metadata.instance.group_id == group.id and self._is_expired_orphan_instance(metadata, timeout):
+                    group_id_to_instance_ids[group.id].append(metadata.instance.instance_id)
```

Comment:
do we need to iterate over the groups? would something like 
```suggestion
        for metadata in node_metadatas:
            if self._is_expired_orphan_instance(metadata, timeout):
              
  group_id_to_instance_ids[metadata.instance.group_id].append(metadata.instance.instance_id)
```

if we want to make sure that the group id is one we care about we could make a set with all those ids and do something like `if metadata.instance.group_id in known_groups` though - instead of nesting loops